frontend/src/components/WaitingRoom/PrecallNetworkTestDialog/PrecallNetworkTestDialog.tsx

```diff
@@ -0,0 +1,372 @@
+ *   @property {boolean} isPrecallNetworkTestOpen - Whether the dialog is open.
+ *   @property {Function} setIsPrecallNetworkTestOpen - Function to set the open state of the dialog.
+ * @returns {ReactElement} The pre-call network test dialog component.
+ */
+const PrecallNetworkTestDialog = ({
+  isPrecallNetworkTestOpen,
+  setIsPrecallNetworkTestOpen,
+}: PrecallNetworkTestDialogProps): ReactElement => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const roomName = useRoomName();
+  const { state, testQuality, stopTest, clearResults } = useNetworkTest();
+  const [hasRunTest, setHasRunTest] = useState(false);
+
+  const handleClose = () => {
+    stopTest();
+    clearResults();
+    setHasRunTest(false);
+    setIsPrecallNetworkTestOpen(false);
+  };
+
+  const handleRetry = async () => {
+    clearResults();
+    setHasRunTest(false);
+    try {
+      await testQuality(roomName);
+      setHasRunTest(true);
+    } catch (error) {
+      console.error('Network test failed:', error);
+      setHasRunTest(true);
+    }
+  };
+
+  const handleStopTest = () => {
+    stopTest();
+    clearResults();
+    setHasRunTest(false);
+    setIsPrecallNetworkTestOpen(false);
+  };
+
+  useEffect(() => {
+    if (
+      isPrecallNetworkTestOpen &&
+      !state.isTestingQuality &&
+      !state.qualityResults &&
+      !hasRunTest
+    ) {
+      testQuality(roomName)
+        .then(() => {
+          setHasRunTest(true);
+        })
+        .catch((error) => {
+          console.error('Network test failed:', error);
+          setHasRunTest(true);
+        });
+    }
+  }, [
+    isPrecallNetworkTestOpen,
+    state.isTestingQuality,
+    state.qualityResults,
+    hasRunTest,
+    testQuality,
+    roomName,
+  ]);
+
+  const audioScore = state.qualityResults?.audio?.mos
+    ? Math.round(state.qualityResults.audio.mos * 100) / 100
+    : null;
+  const videoScore = state.qualityResults?.video?.mos
+    ? Math.round(state.qualityResults.video.mos * 100) / 100
+    : null;
+
+  const audioSupportTitle = useMemo(() => {
+    if (audioScore === null) return undefined;
+    return t(
+      audioScore >= 3
+        ? 'waitingRoom.precallNetworkTest.audioSupported'
+        : 'waitingRoom.precallNetworkTest.audioNotSupported'
+    );
+  }, [audioScore, t]);
+
+  const videoSupportTitle = useMemo(() => {
+    if (videoScore === null) return undefined;
+    return t(
+      videoScore >= 3
+        ? 'waitingRoom.precallNetworkTest.videoSupported'
+        : 'waitingRoom.precallNetworkTest.videoNotSupported'
+    );
+  }, [videoScore, t]);
+
+  const formatScore = (score: number | null): string => {
+    if (score === null) return '—';
+    return `${score.toFixed(2)}/5`;
+  };
+
+  return (
+    <Dialog open={isPrecallNetworkTestOpen} onClose={handleClose} maxWidth="md" fullWidth>
+      <DialogTitle sx={{ m: 0, p: 0, pl: 3, pt: 3, backgroundColor: theme.colors.surface }}>
+        <Typography
+          component="div"
```

Comment:
1. Semantic HTML:  DialogTitle renders an `<h2>`, but contains a nested Typography with `component="div"`, creating invalid `<h2><div> </h2>`nesting. The Typography could be `component="span"`.

2. Alignment: The `IconButton` has (`absolute` positioned at `top: 16`) and title (`pt: 3`) aren't vertically aligned. 

```suggestion
        <DialogTitle
        sx={{
          m: 0,
          p: 3,
          pb: 0,
          backgroundColor: theme.colors.surface,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
        }}
      >
        <Typography
          component="span"
          variant="h5"
          sx={{
            fontWeight: theme.typography.weight['body-base'].value,
            color: theme.colors.textSecondary,
          }}
        >
          {t('waitingRoom.precallNetworkTest.title')}
        </Typography>
        <IconButton
          aria-label={t('button.close')}
          onClick={handleClose}
          sx={{
            color: theme.colors.secondary,
          }}
        >
          <VividIcon name="close-line" customSize={-5} />
        </IconButton>
      </DialogTitle>
```

Reply:
the code for this was brought over from the `BackgroundEffectsDialog` component. I don't mind changing it but we should probably change it there too at some point to be consistent